mobile/lib/modules/home/ui/asset_grid/asset_grid_data_structure.dart

```diff
@@ -33,85 +35,112 @@ class RenderAssetGridElement {
+    const day_format_key = "day_format";
+    const day_format_year_key = "day_format_year";
+    const groups_key = "groups";
+    const per_row_key = "per_row";
+
+    processData(Map<String, Object> data) {
+      final monthFormat = DateFormat(data[month_format_key]! as String);
+      final dayFormatSameYear = DateFormat(data[day_format_key]! as String);
+      final dayFormatOtherYear =
+          DateFormat(data[day_format_year_key]! as String);
+      final groups = data[groups_key]! as Map<String, List<Asset>>;
+      final perRow = data[per_row_key]! as int;
+
+      List<RenderAssetGridElement> elements = [];
+      DateTime? lastDate;
+
+      groups.forEach((groupName, assets) {
+        try {
+          final date = DateTime.parse(groupName);
+
+          if (lastDate == null || lastDate!.month != date.month) {
+            // Month title
+
+            var monthTitleText = groupName;
+
+            try {
+              monthTitleText = monthFormat.format(DateTime.parse(groupName));
+            } catch (e) {
+              log.severe("Failed to format date for month title: $groupName");
+            }
+
+            elements.add(
+              RenderAssetGridElement(
+                RenderAssetGridElementType.monthTitle,
+                title: monthTitleText,
+                date: date,
+              ),
+            );
+          }
+
+          // Add group title
+          var currentYear = DateTime.now().year;
+          var groupYear = DateTime.parse(groupName).year;
+          var formatDate =
+              currentYear == groupYear ? dayFormatSameYear : dayFormatOtherYear;
+
+          var dateText = groupName;
+
+          try {
+            dateText = formatDate.format(DateTime.parse(groupName));
+          } catch (e) {
+            log.severe("Failed to format date for day title: $groupName");
+          }
+
+          elements.add(
+            RenderAssetGridElement(
+              RenderAssetGridElementType.dayTitle,
+              title: dateText,
+              date: date,
+              relatedAssetList: assets,
+            ),
+          );
+
+          // Add rows
+          int cursor = 0;
+          while (cursor < assets.length) {
+            int rowElements = min(assets.length - cursor, perRow);
+
+            final rowElement = RenderAssetGridElement(
+              RenderAssetGridElementType.assetRow,
+              date: date,
+              assetRow: RenderAssetGridRow(
+                assets.sublist(cursor, cursor + rowElements),
+              ),
+            );
+
+            elements.add(rowElement);
+            cursor += rowElements;
+          }
+
+          lastDate = date;
+        } catch (e, stackTrace) {
+          log.severe(e, stackTrace);
+        }
+      });
+
+      return RenderList(elements);
     }
-  });
 
-  return elements;
+    // Compute only allows for one parameter. Therefore we pass all parameters in a map
+    return compute(processData, {
```

Comment:
I think a file-local,private data class (that simply contains all the relevant fields) would be a lot better than a Map. This gives nice static typing. Also, this removes the `*_key` definitions and no longer requires checks if the value exists in the map